File: src/main/java/net/onelitefeather/pandorascluster/land/ChunkPlaceholder.java
package net.onelitefeather.pandorascluster.land;

import jakarta.persistence.*;
import org.jetbrains.annotations.NotNull;

@Entity
public class ChunkPlaceholder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private Long chunkIndex;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "land_id")
    private Land land;

    public ChunkPlaceholder() {
    }

    public ChunkPlaceholder(@NotNull Long chunkIndex) {
        this.chunkIndex = chunkIndex;
    }

    public ChunkPlaceholder(Long chunkIndex, Land land) {
        this.chunkIndex = chunkIndex;
        this.land = land;
    }

    @NotNull
    public Long getId() {
        return id;
    }

    public void setId(@NotNull Long id) {
        this.id = id;
    }

    @NotNull
    public Long getChunkIndex() {
        return chunkIndex;
    }

    public void setChunkIndex(@NotNull Long chunkIndex) {
        this.chunkIndex = chunkIndex;
    }

    @Override
    public String toString() {
        return "ChunkPlaceholder{" +
                "id=" + id +
                ", chunkIndex=" + chunkIndex +
                '}';
    }

    public void setLand(Land land) {
        this.land = land;
    }

    public Land getLand() {
        return land;
    }
}

File: api/src/main/kotlin/net/onelitefeather/pandorascluster/api/models/database/ChunkPlaceholder.kt
package net.onelitefeather.pandorascluster.api.models.database

import jakarta.persistence.*
import org.hibernate.Hibernate

@Entity
data class ChunkPlaceholder(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column
    val chunkIndex: Long = -1,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "land_id")
    val land: Land? = null
) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as ChunkPlaceholder
        return id == other.id
    }

    override fun hashCode(): Int = javaClass.hashCode()

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(id = $id , chunkIndex = $chunkIndex )"
    }
}